client/src/components/AppMenu.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, MouseEvent } from 'react';
+import { NavLink } from 'react-router-dom';
+import {
+  createStyles,
+  IconButton,
+  makeStyles,
+  Menu,
+  MenuItem,
+  Theme,
+} from '@material-ui/core';
+import { Menu as MenuIcon } from '@material-ui/icons';
+
+export const AppMenu = () => {
+  const classes = useStyles();
+
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);

```

Comment:
Pierwsze widzę takie rozwiązanie - trzymanie HTMLElement w stanie. Domyślam się też, że tracking zmian nie jest tutaj zbyt optymalny - te obiekty nie są małe.
Też niezbyt to czytelne. 

Niezbyt rozumiem ten sposób. Dałoby to się zmienić na jakiś properties?

Reply:
To rozwiązanie jest żywcem wzięte z dokumentacji MUI, konkretnie [stąd](https://material-ui.com/components/menus/). Ale zgadzam się, że nie wygląda to najlepiej, warto się temu przyjrzeć.

Reply:
Już nie raz się przekonałem (przynajmniej na backendzie), że dokumentacja jest źródłem możliwości a nie najlepszych praktyk :)